app/models/nic/interface.rb

```diff
@@ -83,6 +83,10 @@ def fqdn_before_last_save
       name_before_last_save.include?('.') ? name_before_last_save : "#{name_before_last_save}.#{domain_before_last_save}"
     end
 
+    def mac
+      attributes['mac']&.downcase
+    end
```

Comment:
Please create two methods named `mac_downcase` and `mac_upcase` and then add all two methods into safemode whitelist. Then use whatever you need in your template. I would like to avoid change of behavior.

Reply:
The behavior change is expected. Creating `mac_downcase` and `mac_upcase` is useless as we can already use `mac.downcase` and `mac.upcase` in templates.

It seems my issue explanation was not enough clear: my troubles comes from the MAC address change during machine life-cycle due to switch between upcased and downcased.

From iPXE to finish template, the process is disturbed by MAC changes.

Reply:
Let me know if clearer scenario is require to understand the issues.

Reply:
Oh I understand now. Please implement an ActiveRecord validator which converts mac to downcase prior saving the record. If you need to fix all existing records, create new Rails migration which makes sure all mac addresses are in lowercase.

Reply:
Oh we already have that in our codebase: `normalize_mac`

Reply:
Hello, I hope you're doing well. My last comment is blocking this from to be merged.

Reply:
Hello @lzap your advises are perfect. I'm working on another topic and will be back on this one in few weeks.

I'll turn this PR in draft.

Thanks!